airflow/utils/dag_processing.py

```diff
@@ -543,94 +543,86 @@ def wait_until_finished(self):
-                                                        signal_conn,
-                                                        _stat_queue,
-                                                        result_queue,
-                                                        async_mode)
-
-            processor_manager.start()
-
-        p = multiprocessing.Process(target=helper,
-                                    args=(),
-                                    name="DagFileProcessorManager")
-        p.start()
-        return p
+    def _run_processor_manager(dag_directory,
+                               file_paths,
+                               max_runs,
+                               processor_factory,
+                               signal_conn,
+                               async_mode):
+        from setproctitle import setproctitle
+        setproctitle("airflow scheduler -- DagFileProcessorManager")
+        # Reload configurations and settings to avoid collision with parent process.
+        # Because this process may need custom configurations that cannot be shared,
+        # e.g. RotatingFileHandler. And it can cause connection corruption if we
+        # do not recreate the SQLA connection pool.
+        os.environ['CONFIG_PROCESSOR_MANAGER_LOGGER'] = 'True'
+        # Replicating the behavior of how logging module was loaded
+        # in logging_config.py
+        reload_module(import_module(airflow.settings.LOGGING_CLASS_PATH.rsplit('.', 1)[0]))
+        reload_module(airflow.settings)
+        airflow.settings.initialize()
+        del os.environ['CONFIG_PROCESSOR_MANAGER_LOGGER']
+        processor_manager = DagFileProcessorManager(dag_directory,
+                                                    file_paths,
+                                                    max_runs,
+                                                    processor_factory,
+                                                    signal_conn,
+                                                    async_mode)
+
+        processor_manager.start()
 
     def harvest_simple_dags(self):
         """
         Harvest DAG parsing results from result queue and sync metadata from stat queue.
+
         :return: List of parsing result in SimpleDag format.
         """
-        # Metadata and results to be harvested can be inconsistent,
-        # but it should not be a big problem.
-        self._sync_metadata()
-        # Heartbeating after syncing metadata so we do not restart manager
-        # if it processed all files for max_run times and exit normally.
-        self._heartbeat_manager()
+
         simple_dags = []
-        while True:
-            try:
-                result = self._result_queue.get_nowait()
-                try:
-                    simple_dags.append(result)
-                finally:
-                    self._result_queue.task_done()
-            except Empty:
-                break
+        while self._parent_signal_conn.poll():
```

Comment:
So I understand that we use one Pipe to send both Stats and parsing results? Nice.

Reply:
Yes - a single pipe that both(al) types ofmessages get sent over. Again, this is sort of where I approached it as "how would I do it in golang", and running things with `go test --race` has sort of taught me to do it this way.

Reply:
Perfect!